src/services/Game/services/BallMovementChecker.ts

```diff
@@ -0,0 +1,221 @@
+            this.map[underBallBlockY].charAt(underBallBlockX),
+        );
+
+        return result;
+    }
+
+    isBallCanGoLeft(ballX: number, ballY: number): boolean {
+        // eslint-disable-next-line no-param-reassign
+        ballX = Math.round(ballX * 100000000) / 100000000;
+        // eslint-disable-next-line no-param-reassign
+        ballY = Math.round(ballY * 100000000) / 100000000;
+
+        const ballMiddleY = Math.round(
+            (ballY + GameConstants.GAMER_DIAMETR / 3)
+                / GameConstants.PERFECT_ONE,
+        );
+
+        const leftBallX = Math.ceil((ballX - 3) / GameConstants.PERFECT_ONE) - 1;
+
+        const result = this.emptyBlocks.includes(
+            this.map[ballMiddleY].charAt(leftBallX),
+        );
+
+        return result;
+    }
+
+    isBallStuckInLeftWall(ballX: number, ballY: number): boolean {
+        // eslint-disable-next-line no-param-reassign
+        ballX = Math.round(ballX * 100000000) / 100000000;
+        // eslint-disable-next-line no-param-reassign
+        ballY = Math.round(ballY * 100000000) / 100000000;
+
+        const ballCurrentBlockY = Math.floor(ballY / GameConstants.PERFECT_ONE);
+
+        const ballCurrentBlockX = Math.floor(ballX / GameConstants.PERFECT_ONE);
+
+        const result = !this.emptyBlocks.includes(
+            this.map[ballCurrentBlockY].charAt(ballCurrentBlockX),
+        );
+
+        return result;
+    }
+
+    isBallCanGoRight(ballX: number, ballY: number): boolean {
+        // eslint-disable-next-line no-param-reassign
+        ballX = Math.round(ballX * 100000000) / 100000000;
+        // eslint-disable-next-line no-param-reassign
+        ballY = Math.round(ballY * 100000000) / 100000000;
+
+        const ballMiddleY = Math.round(
+            (ballY + GameConstants.GAMER_DIAMETR / 3)
+                / GameConstants.PERFECT_ONE,
+        );
+
+        const rightBallX = Math.ceil(
+            (ballX + GameConstants.GAMER_DIAMETR + 3)
+                    / GameConstants.PERFECT_ONE,
+        ) - 1;
+
+        const result = this.emptyBlocks.includes(
+            this.map[ballMiddleY].charAt(rightBallX),
+        );
+
+        return result;
+    }
+
+    isBallStuckInRightWall(ballX: number, ballY: number): boolean {
+        // eslint-disable-next-line no-param-reassign
+        ballX = Math.round(ballX * 100000000) / 100000000;
+        // eslint-disable-next-line no-param-reassign
+        ballY = Math.round(ballY * 100000000) / 100000000;
+
+        const ballCurrentBlockY = Math.floor(ballY / GameConstants.PERFECT_ONE);
+
+        const ballCurrentBlockX = Math.floor(
+            (ballX + GameConstants.GAMER_DIAMETR) / GameConstants.PERFECT_ONE,
+        );
+
+        const result = !this.emptyBlocks.includes(
+            this.map[ballCurrentBlockY].charAt(ballCurrentBlockX),
+        );
+
+        return result;
+    }
+
+    private convertCurrentBlockType(blockSymbol: string): void {
```

Comment:
❗
По смыслу это тоггл
```suggestion
    private toggleCurrentBlockType(blockSymbol: string): void {
```

Reply:
Renamed